lib/twitter_cldr/timezones/location.rb

```diff
@@ -14,6 +14,12 @@ def tz_id
       def resource
         @resource ||= TwitterCldr.get_locale_resource(tz.locale, :timezones)[tz.locale]
       end
+
+      private
+
+      def get_period_for_naming(date)
+        tz.orig_tz.period_for_utc(date.utc)
```

Comment:
Hmm, I don't think we want to get the period for only the UTC version of the date. Although the variable in the code is called `date`, TwitterCLDR supports formatting `Date`, `Time`, and `DateTime` instances, meaning `date` could be any one of those types and could therefore contain a time component. Converting the date to UTC first means ignoring that time component.

Reply:
Hm, TZInfo doesn't support passing a `Date` to `period_for_local`.  I could do `date.to_time.utc` which would handle if a `Date` was passed but it looks like in the original version passing a `Date` wouldn't have worked.

More to your original point though, I could just roll these changes back and simply add options to force the caller to handle it if you think that's the best way to handle this issue.

Reply:
Ah sorry you're right, just `Time` and `DateTime` instances then. TwitterCLDR supports formatting `Date` instances, but because there's no time component the time zone logic is never invoked.

Yeah, I think expecting the caller to handle the ambiguity is the right approach.

Reply:
Sounds like a plan. I'll tinker with it a bit over the weekend.